p4-16/spec/P4-16-spec.mdk

```diff
@@ -1931,6 +1931,202 @@ described below.
 All operations that can be performed on infinite-precision integers
 are described in Section [#sec-varint-ops].
 
+#### Saturating integers
+
+This integer data type enables performing saturating arithmetic, for
+bot signed and unsigned integer types.  All operations such as
+addition and multiplication are limited to a fixed range between a
+minimum and maximum value. Saturating arithmetic has advantages, in
+particular when used as counters. The the result of a saturating
+counter max-ing out is much closer to the real result than a counter
+that overflows and wraps around. According to Wikipedia
+(https://en.wikipedia.org/wiki/Saturation_arithmetic) saturating
+arithmetic is as numerically close to the true answer as possible; for
+8-bit binary signed arithmetic, when the correct answer is 130, it is
+considerably less surprising to get an answer of 127 from saturating
+arithmetic than to get an answer of −126 from modular arithmetic.
+Likewise, for 8-bit binary unsigned arithmetic, when the correct
+answer is 258, it is less surprising to get an answer of 255 from
+saturating arithmetic than to get an answer of 2 from modular
+arithmetic.
+
+P4 supports two data types for saturating arithmetic in P4_16,
+`saturating_bit<W>` and `saturating_int<W>`, for unsigned and signed
```

Comment:
While bit<n> sounds OK and is, indeed representing the idea of a bit string first and its interpretation as an unsigned integer second, the words "saturating_bit" simply do not make a lot of sense to me, since bits do not generally saturate.